python/py-pygls/Portfile

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-pygls
+version             0.11.3
+revision            0
+
+supported_archs     noarch
+license             Apache-2
+maintainers         {judaew @judaew} openmaintainer
+
+description         \
+    Pythonic generic implementation of the Language Server Protocol
+long_description    \
+    pygls (pronounced like \"pie glass\") is a pythonic generic implementation \
+    of the Language Server Protocol for use as a foundation for writing \
+    language servers using Python (e.g. Python, XML, etc.). It allows you to \
+    write your own language server in just a few lines of code.
+homepage            https://github.com/openlawlibrary/pygls
+
+checksums           rmd160  839f8f9255e891d44a21d9f821f419c260fc090e \
+                    sha256  4d86fc854e6d6613cd42bf7511e9c6aac947fc8d62ff973a705570b036d969f2 \
+                    size    139426
+
+python.versions     38 39 310
+
+if {${name} ne ${subport}} {
+    depends_build-append    port:py${python.version}-setuptools \
+                            port:py${python.version}-setuptools_scm \
+                            port:py${python.version}-toml \
+                            port:py${python.version}-wheel
```

Comment:
since upstream provides a `pyproject.toml` file you'd ideally transition building this port to use `python.pep517` yes